pkg/apis/v1/nodeclaim.go

```diff
@@ -22,8 +22,8 @@ import (
 	"k8s.io/apimachinery/pkg/runtime"
 )
 
-// NodeClaimSpec describes the desired state of the NodeClaim
-type NodeClaimSpec struct {
+// NodeClaimCoreSpec describes the shared fields between NodeClaimTemplate and NodeClaimSpec
```

Comment:
Can we just keep the same go struct types and then dynamically remove the resources field from the openAPI spec?

Reply:
Okay, I updated the PR to reflect that